public/app/features/browse-dashboards/api/browseDashboardsAPI.ts

```diff
@@ -69,11 +69,41 @@ function createBackendSrvBaseQuery({ baseURL }: { baseURL: string }): BaseQueryF
   return backendSrvBaseQuery;
 }
 
+// TODO: put this somewhere better :)
+export interface ListFolderArgs {
+  page: number;
+  parentUid: string | undefined;
+  limit: number;
+}
+
 export const browseDashboardsAPI = createApi({
   tagTypes: ['getFolder'],
   reducerPath: 'browseDashboardsAPI',
   baseQuery: createBackendSrvBaseQuery({ baseURL: '/api' }),
   endpoints: (builder) => ({
+    listFolders: builder.query<FolderDTO[], ListFolderArgs>({
+      // TODO: provides tags
+      query: ({ page, parentUid, limit }) => ({ url: '/folders', params: { page, parentUid, limit } }),
```

Comment:
Turns out if we just remember the arguments we used for the requests, we can just select them each individually from the store, and don't need to merge them together.